src/components/views/ViewCampaign.jsx

```diff
@@ -81,7 +83,7 @@ const ViewCampaign = ({ match }) => {
       loadFromScratch ? 0 : aggregateDonations.length,
       (_donations, _donationsTotal) => {
         setAggregateDonations(loadFromScratch ? _donations : aggregateDonations.concat(_donations));
```

Comment:
Array are mutable data types, so concatenating `aggregateDonations` doesn't update the reference. 
```suggestion
        setAggregateDonations(loadFromScratch ? _donations : [...aggregateDonations, ..._donations]);
```

Reply:
Since `loadFromScratch` is `true`, it is not concatenating.